### ISSUE TITLE
babel-preset-env and `export * from './meta'`

### ISSUE THREAD

**Issue by haf** [babel/babel#6951]
babel-preset-es2015 worked just fine, but this new preset doesn't.

It complains about a module exportDefaultFrom (perhaps it's the linter complaining), but it's also crashing on compilation.

And `exportDefaultFrom` is not possible to google in this context; hence this is a bug.

**Comment by hzoo** [comment]
`export * as meta from './meta'` isn't part of es2015, so not sure what you mean here? You'll need to provide some more info

**Comment by haf** [comment]
I have these packages:

```

    "babel-cli": "^6.26.0",
    "babel-core": "^6.26.0",
    "babel-eslint": "^8.0.2",
    "babel-loader": "^7.1.2",
    "babel-plugin-add-module-exports": "^0.2.1",
    "babel-plugin-flow-react-proptypes": "^9.1.1",
    "babel-plugin-react-intl": "^2.3.1",
    "babel-plugin-react-transform": "^3.0.0",
    "babel-plugin-transform-react-constant-elements": "^6.23.0",
    "babel-plugin-transform-runtime": "^6.23.0",
    "babel-plugin-webpack-alias": "^2.1.2",
    "babel-polyfill": "^6.26.0",
    "babel-preset-es2015": "^6.24.1",
    "babel-preset-flow": "^6.23.0",
    "babel-preset-react": "^6.24.1",
    "babel-preset-react-hmre": "^1.1.1",
    "babel-preset-stage-0": "^6.24.1",
    "babel-runtime": "^6.26.0",
    "bundle-loader": "^0.5.5",
```

And they compile code like:

```
import * as things from './things'
export * from './things'
```

But with the latest preset, it doesn't.

**Comment by loganfsmyth** [comment]
I think we need more information. What specifically are you seeing? If you get an error, post the full error output.

**Comment by haf** [comment]
```
/index.js
  2:10  error  Parsing error: This experimental syntax requires enabling the parser plugin: 'exportNamespaceFrom'

  1 | // @flow
> 2 | export * as meta from './meta';
    |          ^
  3 | export { default as List } from './List';
  4 | export { default as Edit } from './Edit';
  5 | export { create } from './actions';

```

**Comment by haf** [comment]
```
ERROR in ./index.js
Module build failed: Error: Failed to load plugin exportNamespaceFrom: Cannot find module 'eslint-plugin-exportNamespaceFrom'
    at Function.Module._resolveFilename (module.js:469:15)
    at Function.resolve (internal/module.js:27:19)
    at Plugins.load (app/node_modules/eslint/lib/config/plugins.js:106:29)
    at Array.forEach (native)
    at Plugins.loadAll (app/node_modules/eslint/lib/config/plugins.js:146:21)
    at loadFromDisk (app/node_modules/eslint/lib/config/config-file.js:492:35)
    at Object.load (app/eslint/lib/config/config-file.js:550:20)
    at Config.loadSpecificConfig (app/node_modules/eslint/lib/config.js:139:46)
    at new Config (app/node_modules/eslint/lib/config.js:101:14)
    at new CLIEngine (app/node_modules/eslint/lib/cli-engine.js:420:23)
    at Object.module.exports (app/node_modules/eslint-loader/index.js:176:27)
 @ multi babel-polyfill event-source-polyfill whatwg-fetch ./index
```

**Comment by loganfsmyth** [comment]
@existentialism This must be a regression from your change. Looks like `babel-eslint` depends on a broad range of `babylon` versions instead of locking it down.

The quickfix would be to publish `babel-eslint` with a locked-down `babylon`. I'd vote for that.

**Comment by haf** [comment]
Hmm. Not much for me to do then, but to roll back or replace the syntax.

**Comment by loganfsmyth** [comment]
Sorry about that, if you want you can install the previous version of `babel-eslint`. e.g.

    "babel-eslint": "^8.0.2",

to

    "babel-eslint": "8.0.1",

without the `^`.

**Comment by haf** [comment]
I rewrote the exports instead. Will unsubscribe from this issue; do what you wish with it.

**Comment by loganfsmyth** [comment]
Okay, your call. We've also now published `8.0.3` so this should be resolved.

**Comment by haf** [comment]
+1
